src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -260,6 +260,15 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
             return instance;
         }
 
+        [RequiresUnreferencedCode("Cannot statically analyze what the element type is of the object collection in type so its members may be trimmed.")]
+        private static object? BindToSet(Type type, IConfiguration config, BinderOptions options)
+        {
+            Type genericType = typeof(HashSet<>).MakeGenericType(type.GenericTypeArguments[0]);
```

Comment:
I'd assume this would change based on the comment by @halter73 in https://github.com/dotnet/runtime/issues/66141#issuecomment-1075743040